src/modules/integrations/direnv.nix

```
@@ -0,0 +1,25 @@
+{ pkgs, config, lib, ... }:
+
+let
+  cfg = config.direnv;
+in
+{
+  options.direnv = {
+    watchFiles = lib.mkOption {
+      description = lib.mdDoc "Extra files to watch for changes and reload the direnv environment";
+      # TODO: best type here? any way to exclude defaults?
+      # TODO: regex support?
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
+    };
+
+    watchfiles = lib.mkOption {
+      internal = true;
+      type = lib.types.package;
+    };
+  };
+
+  config = {
+    direnv.watchfiles = pkgs.writeText "watchfiles.txt" (lib.concatStringsSep "\n" cfg.watchFiles);
+  };
```

Pretty new to nix, does not work also?
```suggestion
  cfg.watchfiles = pkgs.writeText "watchfiles.txt" (lib.concatStringsSep "\n" cfg.watchFiles);
```

---

Nope, the module system expects a top-level `config` attrset that it then merges together. This is setting `watchfiles` on an attrset literally called `cfg`, which doesn't exist.